### PR TITLE
refactor deletequery, removed render prop on query

### PR DESCRIPTION
**Pull request by mlafontant**

### CODE REVIEW

File: src/components/GraphiQL.js [hunk 1]
```diff
@@ -815,79 +784,78 @@ export class GraphiQL extends React.Component {
...
-    let renderAndEmpty = false;
-    for (let i = 0; i < this.state.queryList.length; i += 1) {
-      if (this.state.queryList[i].render && !this.state.queryList[i].query) {
-        renderAndEmpty = true; // would hit this flag if there was no value in the query editor
-      }
-    }
+    //Check if one of the boxes are already empty, return the index of the first empty box; stays -1 if none are empty
+    let emptyIdx;
+    this.state.queryList.forEach((box, i) => {
+      emptyIdx = !box.query ? i : emptyIdx;
```
Comment (mlafontant) [inline src/components/GraphiQL.js]: emptyIdx = !box.query ? i : null;

File: src/components/GraphiQL.js [hunk 2]
```diff
@@ -815,79 +784,78 @@ export class GraphiQL extends React.Component {
...
-        if (queryListCopy[i].render && queryListCopy[i].query === "") {
-          queryListCopy[i].query = query;
-          queryListCopy[i].checked = true;
-        }
-      }
-      this.setState({ queryList: queryListCopy });
-    }
-    // and would not enter into this if statement's code block
-    if (!renderAndEmpty) {
+    if (!!query && emptyIdx !== undefined)
```
Comment (mlafontant) [inline src/components/GraphiQL.js]: if (!!query && emptyIdx !== null)

File: src/components/GraphiQL.js [hunk 3]
```diff
@@ -815,79 +784,78 @@ export class GraphiQL extends React.Component {
...
+      this.setState((prevState, props) => {
+        prevState.queryList[emptyIdx].query = query;
+        return {
+          queryList: prevState.queryList
+        };
+      });
+
+    // if there are no empty boxes, add one
+    if (emptyIdx === undefined) {
```
Comment (mlafontant) [inline src/components/GraphiQL.js]: +    if (emptyIdx === null) {

File: src/components/GraphiQL.js [hunk 4]
```diff
@@ -815,79 +784,78 @@ export class GraphiQL extends React.Component {
...
-    for (let i = 0; i < this.state.queryList.length; i += 1) {
-      if (this.state.queryList[i].render) {
-        renders += 1;
-      }
+    if (idX) {
+      setState((prevState, props) => {
+        // find the query with the same id as the button
+        prevState.queryList[idX].checked = isChecked;
+        return {
+          queryList: prevState
```
Comment (mlafontant) [inline src/components/GraphiQL.js]: queryList: prevState.queryList

Comment (mlafontant) [reply src/components/GraphiQL.js]: Once the id is found, perform operation

Comment (mlafontant) [reply src/components/GraphiQL.js]: use for Loop so that we can break out of it once the id is found and operation is performed

**Comment by eduardomaillo** [comment]
Nice refactorization.

**Pull request merged**